client/src/app/queries/settings.ts

```diff
@@ -0,0 +1,41 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+import { getSettingById, updateSetting } from "@app/api/rest";
+import {
+  Setting,
+  FlagSettingKey,
+  BundleOrderSettingKey,
+} from "@app/api/models";
+
+export const SettingQueryKey = "setting";
+
+export const useSetting = <T>(key: BundleOrderSettingKey | FlagSettingKey) => {
+  const queryClient = useQueryClient();
+
+  if (key === "ui.bundle.order") {
+    return useQuery({
+      queryKey: [SettingQueryKey, key],
+      queryFn: () => getSettingById<T>(key),
+      onError: (error) => console.log(error),
+    });
+  }
+  return useQuery({
+    queryKey: [SettingQueryKey, key],
+    queryFn: () => getSettingById<T>(key),
+    onError: (error) => console.log(error),
+  });
+};
```

Comment:
Both sides of this `if` are the same, and the `queryClient` is never used?

Reply:
@mturley, not the same since we have a return statement either way.
Nice catch re `queryClient`, we don't need it here.

Reply:
> not the same since we have a return statement either way.
I don't understand what you mean. How would this code behave any differently if you removed the `if (key === "ui.bundle.order")` and just left one copy of the `return useQuery({` block? I don't see any differences between these two `useQuery` calls.